bundles/org.openhab.core.io.rest.voice/src/main/java/org/eclipse/smarthome/io/rest/voice/internal/VoiceResource.java

```diff
@@ -169,6 +183,91 @@ public Response interpret(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @ApiParam(va
         }
     }
 
+    @GET
+    @Path("/voices")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Get the list of all voices.", response = VoiceDTO.class, responseContainer = "List")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK") })
+    public Response getVoices() {
+        Collection<Voice> voices = voiceManager.getAllVoices();
+        List<VoiceDTO> dtos = new ArrayList<>(voices.size());
+        for (Voice voice : voices) {
+            dtos.add(VoiceMapper.map(voice));
+        }
+        return Response.ok(dtos).build();
+    }
+
+    @GET
+    @Path("/defaultvoice")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Gets the default voice.", response = VoiceDTO.class)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 404, message = "No default voice was found.") })
+    public Response getDefaultVoice() {
+        Voice voice = voiceManager.getDefaultVoice();
+        if (voice != null) {
+            VoiceDTO dto = VoiceMapper.map(voice);
+            return Response.ok(dto).build();
+        } else {
+            return JSONResponse.createErrorResponse(Status.NOT_FOUND, "Default voice not found");
+        }
+    }
+
+    @POST
+    @Path("/say")
+    @Consumes(MediaType.TEXT_PLAIN)
+    @ApiOperation(value = "Speaks a given text with the default voice through the default audio sink.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 404, message = "No default voice or no default audio sink was found.") })
+    public Response say(@ApiParam(value = "text to speak", required = true) String text) {
+        Voice voice = voiceManager.getDefaultVoice();
+        AudioSink sink = audioManager.getSink();
+        if (voice == null) {
+            return JSONResponse.createErrorResponse(Status.NOT_FOUND, "Default voice not found");
+        } else if (sink == null) {
+            return JSONResponse.createErrorResponse(Status.NOT_FOUND, "Default audio sink not found");
+        } else {
+            voiceManager.say(text, voice.getUID(), sink.getId());
+            return Response.ok(null, MediaType.TEXT_PLAIN).build();
+        }
+    }
+
+    @POST
+    @Path("/say/{sinkid: [a-zA-Z_:0-9]*}")
```

Comment:
The sinkId is not a member of an entity "say", so it should imho rather be a QueryParam instead of a PathParam here.

Reply:
Makes sense.
I think I do it like that because the interpret API  just above was using this pattern.

Reply:
Right, but for the interpreters call, it is selecting an entity by id from the list of interpreters, so it is a different situation.

Reply:
I don't understand the difference.
Here I want to select a voice amongst a list of voices, and a sink amongst a list of audio sinks.

Reply:
Your uri is `say/sinkid` and not `sinks/sinkid` or `voices/voiceid`. Hence "sinkid" is not an entity of the "say" entities. See the difference?

Reply:
I see what you mean now.